### PR TITLE
US26136: Update in game notifications.

### PR DESCRIPTION
**Pull request by jmargh**
This updates the in game notification system, porting it to lua. It implements it with the new gui style, with some changes to how notifications are displayed. All new features are wrapped in fast flags.

### CODE REVIEW

**Comment by einsteinK** [comment]
Nothing important, but it seems when unchecking the "Auto" option for VideoQuality,
there could be a notification appearing.
(Due the new level then being > 0, allowing for a notification)
Seems to differentiate from the old system, which used some weird magic to counter this.

File: CoreScriptsRoot/CoreScripts/NotificationScript2.lua [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+	end
+end
+BadgeService.BadgeAwarded:connect(onBadgeAwarded)
+
+--[[ Graphics Changes Notification ]]--
+GameSettings.Changed:connect(function(property)
+	if property == "SavedQualityLevel" then
+		local level = GameSettings.SavedQualityLevel.Value
+		-- value of 0 is automatic, we do not want to send a notification in that case
+		if level > 0 and level ~= CurrentGraphicsQualityLevel then
```
Comment (jmargh) [inline CoreScriptsRoot/CoreScripts/NotificationScript2.lua]: The case for SavedQualityLevel being 0 is handled here.

**Comment by jmargh** [comment]
https://github.com/ROBLOX/Core-Scripts/blob/master/CoreScriptsRoot/CoreScripts/NotificationScript2.lua#L343

This handles both the cases if it's set to auto and coming back to manual.
Also, thanks for pointing out the previous issues =)

**Comment by einsteinK** [comment]
Ah, I totally ignored the fact that CurrentGraphicsQualityLevel doesn't get set for Auto.

I was bored, and just felt like checking the new script out.

**Pull request merged**